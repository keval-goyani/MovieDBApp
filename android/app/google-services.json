{
  "project_info": {
    "project_number": "89996564819",
    "project_id": "moviedb-2908d",
    "storage_bucket": "moviedb-2908d.appspot.com"
  },
  "client": [
    {
      "client_info": {
        "mobilesdk_app_id": "1:89996564819:android:e942630a87714b1d3ea9a6",
        "android_client_info": {
          "package_name": "com.moviedbapp"
        }
      },
      "oauth_client": [
        {
          "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
          "client_type": 3
        }
      ],
      "api_key": [
        {
          "current_key": "AIzaSyBYrM1MC9Go35l8ra748zqBtCRDkrB_Nn4"
        }
      ],
      "services": {
        "appinvite_service": {
          "other_platform_oauth_client": [
            {
              "client_id": "89996564819-hph4f8e5b765gev43v87vj04t9tun4il.apps.googleusercontent.com",
              "client_type": 3
            }
          ]
        }
      }
    }
  ],
  "configuration_version": "1"
}