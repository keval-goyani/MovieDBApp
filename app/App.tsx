import React, {useEffect} from 'react';
import {SafeAreaView, Text} from 'react-native';
import SplashScreen from 'react-native-splash-screen';
import {styles} from './theme';

const App = () => {
  useEffect(() => {
    SplashScreen.hide();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Text>Hello World</Text>
    </SafeAreaView>
  );
};

export default App;
