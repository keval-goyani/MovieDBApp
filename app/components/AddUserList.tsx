import { useFocusEffect, useNavigation } from '@react-navigation/native';
import React, { useEffect, useState } from 'react';
import { FlatList, Text, TouchableOpacity, View } from 'react-native';
import Check from 'react-native-vector-icons/AntDesign';
import { useDispatch, useSelector } from 'react-redux';
import { asMutable } from 'seamless-immutable';
import { ProfileImage, UserListEmpty } from '../components';
import {
  AddUserListProps,
  NavigationDataType,
  navigationStrings,
  RenderItemTypes,
  SelectedUsersProps,
  strings,
  UserDataType,
} from '../constants';
import { authDataSelectors } from '../redux/AuthRedux';
import { userListSelector } from '../redux/UserListRedux';
import { conversationIdCreation, sub } from '../services';
import { Color, moderateScale } from '../theme';
import { styles } from './styles/AddUserListStyles';

const AddUsersList = ({ userListData, setSelectedUsers }: AddUserListProps) => {
  const navigation: NavigationDataType = useNavigation();
  const dispatch = useDispatch();
  const { user } = useSelector(authDataSelectors.getData);
  const { fetchingUserList } = useSelector(userListSelector.getData);
  const [selectedItems, setSelectedItems] = useState<UserDataType[]>([]);

  useFocusEffect(() => {
    sub({ user, dispatch });
    return () => sub;
  });

  useEffect(() => {
    setSelectedUsers(selectedItems);
  }, [selectedItems, setSelectedUsers]);

  const renderUserList = ({ item, index }: RenderItemTypes) => {
    const usersEmail = [user?.email ?? '', item?.email];
    const conversationId: string = conversationIdCreation(usersEmail);

    const handleOnPress = (selectedUser: UserDataType) => {
      console.log(selectedItems, '<==selectedItems.length');
      console.log(selectedUser, '<==selectedUser');

      if (selectedItems.length) {
        return selectUsers(selectedUser);
      }
      navigation.goBack();
      navigation.navigate(navigationStrings.Chat, {
        conversationId,
        username: item?.username,
        receiverId: item?.uid,
        userStatus: item?.status,
        profileImage: item?.profileImage,
        userEmail: item?.email,
      });
    };

    const getSelected = (usersData: SelectedUsersProps) => {
      return JSON.stringify(selectedItems).includes(JSON.stringify(usersData));
    };

    const selected = getSelected(item);

    const selectUsers = (selectedUser: UserDataType) => {
      if (
        JSON.stringify(selectedItems).includes(JSON.stringify(selectedUser))
      ) {
        const filteredList = selectedItems?.filter(
          (items: UserDataType) => items !== selectedUser,
        );
        setSelectedItems(filteredList);
      } else {
        setSelectedItems([...selectedItems, selectedUser]);
      }
    };

    return (
      <TouchableOpacity
        style={styles.listItem}
        onPress={() => handleOnPress(item)}x
        activeOpacity={0.8}
        onLongPress={() => selectUsers(item)}
        testID={`button-for-test${index}`}>
        <View style={styles.itemContainer}>
          <ProfileImage
            profileImage={item?.profileImage}
            userStatus={item?.status}
          />
          <View style={styles.nameView}>
            <Text style={styles.text}>{item?.username}</Text>
            <Text
              style={styles.lastChatText}
              ellipsizeMode="tail"
              numberOfLines={1}>
              {item?.email}
            </Text>
          </View>
        </View>
        {selected && (
          <>
            <View style={styles.overlay} />
            <Check
              testID="check"
              name={strings.checkIcon}
              size={moderateScale(23)}
              color={Color.lightBlue}
              style={styles.iconStyle}
            />
          </>
        )}
      </TouchableOpacity>
    );
  };

  return (
    <FlatList
      testID={'flatList'}
      data={asMutable(userListData)}
      renderItem={renderUserList}
      ListEmptyComponent={
        <UserListEmpty
          fetching={fetchingUserList}
          userListLength={userListData.length}
        />
      }
      showsVerticalScrollIndicator={false}
      bounces={false}
    />
  );
};

export default AddUsersList;
