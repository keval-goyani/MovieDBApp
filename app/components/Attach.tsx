import { useNavigation } from '@react-navigation/native';
import React, { FC } from 'react';
import { Alert, View } from 'react-native';
import SystemSetting from 'react-native-system-setting';
import { CustomIconRounder } from '../components';
import {
  AttachDataType,
  NavigationDataType,
  navigationStrings,
  strings,
} from '../constants';
import {
  handleCameraPermission,
  handleDocumentPermission,
  handleGalleryPermission,
} from '../services';
import { Icons } from '../theme';
import styles from './styles/AttachStyles';

const Attach: FC<AttachDataType> = ({
  setIsAttach,
  setImagePath,
  setDocumentData,
  conversationId,
  username,
  receiverId,
}) => {
  const navigation: NavigationDataType = useNavigation();

  const turnOnLocation = (enable: boolean) => {
    if (enable) {
      navigation.navigate(navigationStrings.Location, {
        isFromChat: false,
        ...{ conversationId, username, receiverId },
      });
      setIsAttach(false);
    } else {
      Alert.alert(strings.turnOn, strings.goToSetting, [
        {
          onPress: () => SystemSetting.switchLocation(),
          text: strings.setting,
        },
        { text: strings.cancel },
      ]);
    }
  };

  const locationHandler = () => {
    SystemSetting.isLocationEnabled()
      .then(enable => {
        turnOnLocation(enable);
      })
      .catch(error => Alert.alert(strings.warning, error));
  };

  return (
    <View style={styles.container}>
      <View style={styles.iconContainer}>
        <CustomIconRounder
          backgroundColor={styles.documentBackgroundColor}
          path={Icons.fileIcon}
          iconName={strings.document}
          tintColor={styles.documentTintColor}
          onPress={() => {
            setIsAttach(false);
            handleDocumentPermission(setDocumentData);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.cameraBackgroundColor}
          path={Icons.cameraIcon}
          iconName={strings.camera}
          tintColor={styles.cameraTintColor}
          onPress={() => {
            handleCameraPermission(setImagePath);
            setIsAttach(false);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.galleryBackgroundColor}
          path={Icons.galleryIcon}
          iconName={strings.gallery}
          tintColor={styles.galleryTintColor}
          onPress={() => {
            handleGalleryPermission(setImagePath);
            setIsAttach(false);
          }}
        />
        <CustomIconRounder
          backgroundColor={styles.locationBackgroundColor}
          path={Icons.locationIcon}
          iconName={strings.map}
          tintColor={styles.locationTintColor}
          onPress={locationHandler}
        />
      </View>
    </View>
  );
};

export default Attach;
