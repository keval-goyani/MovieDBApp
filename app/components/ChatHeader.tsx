import { useNavigation } from '@react-navigation/native';
import React, { useState } from 'react';
import {
  Image,
  Pressable,
  PressableStateCallbackType,
  Text,
  TouchableOpacity,
  View,
} from 'react-native';
import { ChatMenu, ProfileImage } from '../components';
import {
  appConstants,
  ChatHeaderDataType,
  NavigationDataType,
  navigationStrings,
} from '../constants';
import { Color, Icons } from '../theme';
import { styles } from './styles/ChatHeaderStyles';

const ChatHeader = ({
  username,
  profileImage,
  userStatus,
  showMenu,
  setShowMenu,
  setIsAttach,
  setCameraModal,
  setChatWallpaper,
  conversationId,
  membersName,
  groupName,
  userEmail,
  receiverId,
}: ChatHeaderDataType) => {
  const navigation: NavigationDataType = useNavigation();
  const userInfo = userStatus ?? membersName;

  const styleFn = ({ pressed }: PressableStateCallbackType) => [
    {
      backgroundColor: pressed ? Color.transparentGrey : Color.darkBlue,
    },
    styles.userNameAndStatus,
  ];

  return (
    <>
      <View style={styles.container}>
        <TouchableOpacity
          onPress={() => navigation.goBack()}
          testID={'goBackButton'}>
          <Image source={Icons?.backIcon} style={styles.leftIconStyle} />
        </TouchableOpacity>
        <View style={styles.profileView}>
          <View style={styles.profile}>
            <ProfileImage
              {...{ profileImage, userStatus, groupName }}
              isChatHeader={appConstants.trueValue}
            />
            <Pressable
              testID={'profileImagePressable'}
              onPress={() =>
                navigation.navigate(navigationStrings.ProfileInfo, {
                  profileImage,
                  username,
                  userEmail,
                  conversationId,
                  receiverId,
                  groupName,
                })
              }
              style={styleFn}>
              <Text style={styles.username}>{username}</Text>
              <Text numberOfLines={1} style={styles.statusOrMembers}>
                {userInfo}
              </Text>
            </Pressable>
          </View>
        </View>
        <Pressable
          testID={'dotsMenu'}
          onPress={() => {
            setShowMenu(!showMenu);
            setIsAttach(false);
            setCameraModal(false);
          }}>
          <Image source={Icons?.dotMenu} style={styles.dotsMenu} />
        </Pressable>
      </View>
      {showMenu && (
        <ChatMenu {...{ conversationId, setChatWallpaper, setShowMenu }} />
      )}
    </>
  );
};

export default ChatHeader;
