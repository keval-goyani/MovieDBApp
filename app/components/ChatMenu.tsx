import React from 'react';
import { View } from 'react-native';
import { useDispatch } from 'react-redux';
import { CustomButton } from '../components';
import { ChatMenuDataType, strings } from '../constants';
import { clearChat, handleGalleryPermission } from '../services';
import { styles } from './styles/ChatMenuStyles';

const ChatMenu = ({
  conversationId,
  setChatWallpaper,
  setShowMenu,
}: ChatMenuDataType) => {
  const dispatch = useDispatch();

  const changeWallpaper = () => {
    handleGalleryPermission(setChatWallpaper);
    setShowMenu(false);
  };

  return (
    <View style={styles.container} testID={'chat-menu'}>
      <CustomButton
        {...{
          buttonTextStyle: styles.menuItem,
          buttonStyle: styles.menuListItem,
          buttonText: strings.changeWallpaper,
          onPress: changeWallpaper,
        }} 
      />
      <CustomButton
        {...{
          buttonStyle: styles.menuListItem,
          buttonTextStyle: styles.menuItem,
          buttonText: strings.clearChat,
          onPress: () =>
            clearChat({
              setShowMenu,
              conversationId,
              dispatch,
            }),
        }}
      />
    </View>
  );
};

export default ChatMenu;
