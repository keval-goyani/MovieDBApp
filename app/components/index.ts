import ChatHeader from './ChatHeader';
import ChatInput from './ChatInput';
import CustomSkeleton from './CustomSkeleton';
import DropDownMenu from './DropDownMenu';
import Header from './Header';
import List from './List';
import ListContainer, { movieListItem } from './ListContainer';
import Loader from './Loader';
import LoadingState from './LoadingState';
import Message from './Message';
import MessageList from './MessageList';
import Movies from './Movies';
import MovieTrailer, { trailerListItem } from './MovieTrailers';
import SearchModal from './SearchModal';
import SkeletonCard from './SkeletonCard';
import TabBarIcon from './TabBarIcon';
import Trailers from './Trailers';
import UserListEmpty from './UserListEmpty';
import UsersList from './UsersList';

export {
  ChatHeader,
  ChatInput,
  DropDownMenu,
  Header,
  List,
  ListContainer,
  MovieTrailer,
  Message,
  MessageList,
  TabBarIcon,
  SearchModal,
  Loader,
  LoadingState,
  Movies,
  movieListItem,
  Trailers,
  trailerListItem,
  UsersList,
  SkeletonCard,
  CustomSkeleton,
  UserListEmpty,
};
