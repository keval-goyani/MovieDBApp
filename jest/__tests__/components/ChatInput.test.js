import { fireEvent, render } from '@testing-library/react-native';
import React, { useState } from 'react';
import { ChatInput } from '../../../app/components';
import { Keyboard } from 'react-native';

const setCameraModal = jest.fn();
const setIsAttach = jest.fn();
const setShowMenu = jest.fn();
Keyboard.dismiss = jest.fn();
const isAttach = true;
const cameraModal = true;

const documentData = {
  documentName: 'abd',
  documentUrl: 'http://document.location.',
};

describe('ChatInpur', () => {
  beforeEach(() => {
    useState.mockImplementation(jest.requireActual('react').useState);
  });

  it('take SnapShot of chatInput', () => {
    const { toJSON } = render(<ChatInput />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should setMessage on onchangeText', () => {
    const { getByTestId } = render(<ChatInput />);
    const textInput = getByTestId('textInput');
    fireEvent.changeText(textInput, 'chandani');
    expect(textInput.props.value).toBe('chandani');
  });

  it('should setcamera modal and isAttach on onfocus', () => {
    const { getByTestId } = render(
      <ChatInput setCameraModal={setCameraModal} setIsAttach={setIsAttach} />,
    );
    const textInput = getByTestId('textInput');
    fireEvent(textInput, 'onFocus');
    expect(setCameraModal).toHaveBeenCalledWith(false);
    expect(setIsAttach).toHaveBeenCalledWith(false);
  });

  it('OnPress event of Attach icon', () => {
    const { getByTestId } = render(
      <ChatInput
        setCameraModal={setCameraModal}
        setIsAttach={setIsAttach}
        setShowMenu={setShowMenu}
        isAttach={isAttach}
      />,
    );
    const atttachIcon = getByTestId('AttachIcon');
    fireEvent.press(atttachIcon);
    expect(setCameraModal).toHaveBeenCalledWith(false);
    expect(setShowMenu).toHaveBeenCalledWith(false);
    expect(setIsAttach).toHaveBeenCalledWith(!isAttach);
    expect(Keyboard.dismiss).toHaveBeenCalled();
  });

  it('OnPress event of Camera', () => {
    const { getByTestId } = render(
      <ChatInput
        setCameraModal={setCameraModal}
        setIsAttach={setIsAttach}
        setShowMenu={setShowMenu}
        cameraModal={cameraModal}
      />,
    );
    const cameraIcon = getByTestId('CameraIcon');
    fireEvent.press(cameraIcon);
    expect(setCameraModal).toHaveBeenCalledWith(!cameraModal);
    expect(setIsAttach).toHaveBeenCalledWith(false);
    expect(setShowMenu).toHaveBeenCalledWith(false);
    expect(Keyboard.dismiss).toHaveBeenCalled();
  });

  it('Onpress of send Button', () => {
    const { getByTestId } = render(
      <ChatInput
        setCameraModal={setCameraModal}
        setIsAttach={setIsAttach}
        setShowMenu={setShowMenu}
        imageUrl="chandani.png"
        documentData={documentData}
      />,
    );
    const sendButton = getByTestId('sendButton');
    fireEvent.press(sendButton);
    expect(setCameraModal).toHaveBeenCalledWith(false);
    expect(setIsAttach).toHaveBeenCalledWith(false);
    expect(setShowMenu).toHaveBeenCalledWith(false);
  });

  it('should send message when is there', () => {
    const { getByTestId } = render(
      <ChatInput
        setCameraModal={setCameraModal}
        setIsAttach={setIsAttach}
        setShowMenu={setShowMenu}
      />,
    );
    const textInput = getByTestId('textInput');
    fireEvent.changeText(textInput, 'chandani');
    expect(textInput.props.value).toBe('chandani');
    const sendButton = getByTestId('sendButton');
    fireEvent.press(sendButton);
    expect(setIsAttach).toHaveBeenCalled();
  });
});
