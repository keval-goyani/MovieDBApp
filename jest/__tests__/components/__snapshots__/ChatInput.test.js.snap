// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChatInpur take SnapShot of chatInput 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginHorizontal": undefined,
        "paddingHorizontal": undefined,
        "paddingVertical": undefined,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#f0f0f0",
          "borderRadius": 30,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginRight": undefined,
          "paddingVertical": undefined,
        }
      }
    >
      <TextInput
        multiline={true}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="Type Something..."
        placeholderTextColor="#A9A9A9"
        style={
          Object {
            "backgroundColor": "transparent",
            "color": "#000000",
            "flex": 1,
            "fontSize": undefined,
            "height": undefined,
            "maxHeight": undefined,
            "paddingLeft": undefined,
          }
        }
        testID="textInput"
        value=""
      />
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="AttachIcon"
      >
        <Image
          source={
            Object {
              "default": Object {
                "attach": Object {
                  "testUri": "../../../app/assets/icons/attach.png",
                },
                "avatar": Object {
                  "testUri": "../../../app/assets/icons/user.png",
                },
                "backIcon": Object {
                  "testUri": "../../../app/assets/icons/backIcon.png",
                },
                "camera": Object {
                  "testUri": "../../../app/assets/icons/camera.png",
                },
                "cameraIcon": Object {
                  "testUri": "../../../app/assets/icons/cameraIcon.png",
                },
                "chatBackground": Object {
                  "testUri": "../../../app/assets/icons/chatBackground.png",
                },
                "close": Object {
                  "testUri": "../../../app/assets/icons/close.png",
                },
                "comingSoon": Object {
                  "testUri": "../../../app/assets/icons/comingSoon.png",
                },
                "communityIcon": Object {
                  "testUri": "../../../app/assets/icons/communityIcon.png",
                },
                "document": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "documentIcon": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "dotMenu": Object {
                  "testUri": "../../../app/assets/icons/dots.png",
                },
                "expandIcon": Object {
                  "testUri": "../../../app/assets/icons/expandIcon.png",
                },
                "fileIcon": Object {
                  "testUri": "../../../app/assets/icons/fileIcon.png",
                },
                "fullStop": Object {
                  "testUri": "../../../app/assets/icons/fullStop.png",
                },
                "galleryIcon": Object {
                  "testUri": "../../../app/assets/icons/galleryIcon.png",
                },
                "homeIcon": Object {
                  "testUri": "../../../app/assets/icons/homeIcon.png",
                },
                "imageDocument": Object {
                  "testUri": "../../../app/assets/icons/imageIcon.png",
                },
                "location": Object {
                  "testUri": "../../../app/assets/icons/share.png",
                },
                "locationIcon": Object {
                  "testUri": "../../../app/assets/icons/locationIcon.png",
                },
                "logout": Object {
                  "testUri": "../../../app/assets/icons/logout.png",
                },
                "menuIcon": Object {
                  "testUri": "../../../app/assets/icons/menuIcon.png",
                },
                "movieDbIcon": Object {
                  "testUri": "../../../app/assets/icons/movieDBIcon.png",
                },
                "movieDbIcon2x": Object {
                  "testUri": "../../../app/assets/icons/movieDB2x.png",
                },
                "movieDbIcon3x": Object {
                  "testUri": "../../../app/assets/icons/movieDB3x.png",
                },
                "moviesIcon": Object {
                  "testUri": "../../../app/assets/icons/moviesIcon.png",
                },
                "noItemFound": Object {
                  "testUri": "../../../app/assets/icons/noItemFound.png",
                },
                "notFound": Object {
                  "testUri": "../../../app/assets/icons/notFoundIcon.png",
                },
                "pdfDocument": Object {
                  "testUri": "../../../app/assets/icons/pdfDocument 2.png",
                },
                "playIcon": Object {
                  "testUri": "../../../app/assets/icons/playIcon.png",
                },
                "plus": Object {
                  "testUri": "../../../app/assets/icons/plusSign.png",
                },
                "pptDocument": Object {
                  "testUri": "../../../app/assets/icons/pptDocument.png",
                },
                "searchIcon": Object {
                  "testUri": "../../../app/assets/icons/searchIcon.png",
                },
                "send": Object {
                  "testUri": "../../../app/assets/icons/send.png",
                },
                "threeDotIcon": Object {
                  "testUri": "../../../app/assets/icons/threeDotIcon.png",
                },
                "trailerIcon": Object {
                  "testUri": "../../../app/assets/icons/trailerIcon.png",
                },
                "trending": Object {
                  "testUri": "../../../app/assets/icons/trending.png",
                },
                "tv": Object {
                  "testUri": "../../../app/assets/icons/tv.png",
                },
                "videoDocument": Object {
                  "testUri": "../../../app/assets/icons/mp4Document.png",
                },
              },
            }
          }
          style={
            Object {
              "height": undefined,
              "marginRight": undefined,
              "tintColor": "#808080",
              "width": undefined,
            }
          }
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="CameraIcon"
      >
        <Image
          source={
            Object {
              "default": Object {
                "attach": Object {
                  "testUri": "../../../app/assets/icons/attach.png",
                },
                "avatar": Object {
                  "testUri": "../../../app/assets/icons/user.png",
                },
                "backIcon": Object {
                  "testUri": "../../../app/assets/icons/backIcon.png",
                },
                "camera": Object {
                  "testUri": "../../../app/assets/icons/camera.png",
                },
                "cameraIcon": Object {
                  "testUri": "../../../app/assets/icons/cameraIcon.png",
                },
                "chatBackground": Object {
                  "testUri": "../../../app/assets/icons/chatBackground.png",
                },
                "close": Object {
                  "testUri": "../../../app/assets/icons/close.png",
                },
                "comingSoon": Object {
                  "testUri": "../../../app/assets/icons/comingSoon.png",
                },
                "communityIcon": Object {
                  "testUri": "../../../app/assets/icons/communityIcon.png",
                },
                "document": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "documentIcon": Object {
                  "testUri": "../../../app/assets/icons/documentIcon.png",
                },
                "dotMenu": Object {
                  "testUri": "../../../app/assets/icons/dots.png",
                },
                "expandIcon": Object {
                  "testUri": "../../../app/assets/icons/expandIcon.png",
                },
                "fileIcon": Object {
                  "testUri": "../../../app/assets/icons/fileIcon.png",
                },
                "fullStop": Object {
                  "testUri": "../../../app/assets/icons/fullStop.png",
                },
                "galleryIcon": Object {
                  "testUri": "../../../app/assets/icons/galleryIcon.png",
                },
                "homeIcon": Object {
                  "testUri": "../../../app/assets/icons/homeIcon.png",
                },
                "imageDocument": Object {
                  "testUri": "../../../app/assets/icons/imageIcon.png",
                },
                "location": Object {
                  "testUri": "../../../app/assets/icons/share.png",
                },
                "locationIcon": Object {
                  "testUri": "../../../app/assets/icons/locationIcon.png",
                },
                "logout": Object {
                  "testUri": "../../../app/assets/icons/logout.png",
                },
                "menuIcon": Object {
                  "testUri": "../../../app/assets/icons/menuIcon.png",
                },
                "movieDbIcon": Object {
                  "testUri": "../../../app/assets/icons/movieDBIcon.png",
                },
                "movieDbIcon2x": Object {
                  "testUri": "../../../app/assets/icons/movieDB2x.png",
                },
                "movieDbIcon3x": Object {
                  "testUri": "../../../app/assets/icons/movieDB3x.png",
                },
                "moviesIcon": Object {
                  "testUri": "../../../app/assets/icons/moviesIcon.png",
                },
                "noItemFound": Object {
                  "testUri": "../../../app/assets/icons/noItemFound.png",
                },
                "notFound": Object {
                  "testUri": "../../../app/assets/icons/notFoundIcon.png",
                },
                "pdfDocument": Object {
                  "testUri": "../../../app/assets/icons/pdfDocument 2.png",
                },
                "playIcon": Object {
                  "testUri": "../../../app/assets/icons/playIcon.png",
                },
                "plus": Object {
                  "testUri": "../../../app/assets/icons/plusSign.png",
                },
                "pptDocument": Object {
                  "testUri": "../../../app/assets/icons/pptDocument.png",
                },
                "searchIcon": Object {
                  "testUri": "../../../app/assets/icons/searchIcon.png",
                },
                "send": Object {
                  "testUri": "../../../app/assets/icons/send.png",
                },
                "threeDotIcon": Object {
                  "testUri": "../../../app/assets/icons/threeDotIcon.png",
                },
                "trailerIcon": Object {
                  "testUri": "../../../app/assets/icons/trailerIcon.png",
                },
                "trending": Object {
                  "testUri": "../../../app/assets/icons/trending.png",
                },
                "tv": Object {
                  "testUri": "../../../app/assets/icons/tv.png",
                },
                "videoDocument": Object {
                  "testUri": "../../../app/assets/icons/mp4Document.png",
                },
              },
            }
          }
          style={
            Object {
              "height": undefined,
              "marginRight": undefined,
              "tintColor": "#808080",
              "width": undefined,
            }
          }
        />
      </View>
    </View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#0d253f",
          "borderRadius": 50,
          "height": undefined,
          "justifyContent": "center",
          "opacity": 1,
          "width": undefined,
        }
      }
      testID="sendButton"
    >
      <Image
        source={
          Object {
            "default": Object {
              "attach": Object {
                "testUri": "../../../app/assets/icons/attach.png",
              },
              "avatar": Object {
                "testUri": "../../../app/assets/icons/user.png",
              },
              "backIcon": Object {
                "testUri": "../../../app/assets/icons/backIcon.png",
              },
              "camera": Object {
                "testUri": "../../../app/assets/icons/camera.png",
              },
              "cameraIcon": Object {
                "testUri": "../../../app/assets/icons/cameraIcon.png",
              },
              "chatBackground": Object {
                "testUri": "../../../app/assets/icons/chatBackground.png",
              },
              "close": Object {
                "testUri": "../../../app/assets/icons/close.png",
              },
              "comingSoon": Object {
                "testUri": "../../../app/assets/icons/comingSoon.png",
              },
              "communityIcon": Object {
                "testUri": "../../../app/assets/icons/communityIcon.png",
              },
              "document": Object {
                "testUri": "../../../app/assets/icons/documentIcon.png",
              },
              "documentIcon": Object {
                "testUri": "../../../app/assets/icons/documentIcon.png",
              },
              "dotMenu": Object {
                "testUri": "../../../app/assets/icons/dots.png",
              },
              "expandIcon": Object {
                "testUri": "../../../app/assets/icons/expandIcon.png",
              },
              "fileIcon": Object {
                "testUri": "../../../app/assets/icons/fileIcon.png",
              },
              "fullStop": Object {
                "testUri": "../../../app/assets/icons/fullStop.png",
              },
              "galleryIcon": Object {
                "testUri": "../../../app/assets/icons/galleryIcon.png",
              },
              "homeIcon": Object {
                "testUri": "../../../app/assets/icons/homeIcon.png",
              },
              "imageDocument": Object {
                "testUri": "../../../app/assets/icons/imageIcon.png",
              },
              "location": Object {
                "testUri": "../../../app/assets/icons/share.png",
              },
              "locationIcon": Object {
                "testUri": "../../../app/assets/icons/locationIcon.png",
              },
              "logout": Object {
                "testUri": "../../../app/assets/icons/logout.png",
              },
              "menuIcon": Object {
                "testUri": "../../../app/assets/icons/menuIcon.png",
              },
              "movieDbIcon": Object {
                "testUri": "../../../app/assets/icons/movieDBIcon.png",
              },
              "movieDbIcon2x": Object {
                "testUri": "../../../app/assets/icons/movieDB2x.png",
              },
              "movieDbIcon3x": Object {
                "testUri": "../../../app/assets/icons/movieDB3x.png",
              },
              "moviesIcon": Object {
                "testUri": "../../../app/assets/icons/moviesIcon.png",
              },
              "noItemFound": Object {
                "testUri": "../../../app/assets/icons/noItemFound.png",
              },
              "notFound": Object {
                "testUri": "../../../app/assets/icons/notFoundIcon.png",
              },
              "pdfDocument": Object {
                "testUri": "../../../app/assets/icons/pdfDocument 2.png",
              },
              "playIcon": Object {
                "testUri": "../../../app/assets/icons/playIcon.png",
              },
              "plus": Object {
                "testUri": "../../../app/assets/icons/plusSign.png",
              },
              "pptDocument": Object {
                "testUri": "../../../app/assets/icons/pptDocument.png",
              },
              "searchIcon": Object {
                "testUri": "../../../app/assets/icons/searchIcon.png",
              },
              "send": Object {
                "testUri": "../../../app/assets/icons/send.png",
              },
              "threeDotIcon": Object {
                "testUri": "../../../app/assets/icons/threeDotIcon.png",
              },
              "trailerIcon": Object {
                "testUri": "../../../app/assets/icons/trailerIcon.png",
              },
              "trending": Object {
                "testUri": "../../../app/assets/icons/trending.png",
              },
              "tv": Object {
                "testUri": "../../../app/assets/icons/tv.png",
              },
              "videoDocument": Object {
                "testUri": "../../../app/assets/icons/mp4Document.png",
              },
            },
          }
        }
        style={
          Object {
            "height": undefined,
            "tintColor": "#FFFFFF",
            "width": undefined,
          }
        }
      />
    </View>
  </View>
</View>
`;
